test/data/s3/test_s3.py

```diff
@@ -977,7 +977,7 @@ def test_list_recursive_sibling_prefix_filtering(inject_failure_rate):
             s3_setup.put(file_path, f"test content for {file_path}")
 
     with S3(
-        s3root=f"{base_s3root}/{test_prefix}/log/",
```

Comment:
so the issue here is because s3root has consecutive slashes in the middle.. keeping it as `log/` should work